code/Controllers/CMSMain.php

```diff
@@ -462,8 +464,9 @@ public function getSiteTreeFor(
         $childrenMethod = null,
         $numChildrenMethod = null,
         $filterFunction = null,
-        $nodeCountThreshold = 30
+        $nodeCountThreshold = null
     ) {
+        $nodeCountThreshold = is_null($nodeCountThreshold) ? Config::inst()->get(Hierarchy::class, 'node_threshold_total') : $nodeCountThreshold;
```

Comment:
Hierarchy will only provide the default, which can be overridden on a $className basis, hence the suggestion.